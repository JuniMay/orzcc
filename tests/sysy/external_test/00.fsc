// double loop
#include <stdio.h>
int main(){
    const int n = 5;
    const int m = 11; // Need: m >= 2n+1

    int loop = n;
    int x[n] = {};
    int y[n]  = {};
    int i = 1;
    while (i<=n){
        y[i-1] = i;
        i = i+1;
    }
    
    int z[m]  = {};
    i = m;
    int j = 0;
    while (i>0){
        z[j] = i;
        i = i-1;
        j = j+1;
    }

    int q = 1, r = 2, t = 3;

    int l = 1;
    while (l <= loop) {
        int k = 0;
        while (k < n) {
            x[k] = q + y[k] * (r * z[k + 1] + t * z[k + 2]);
            k = k+1;
        }
        l = l+1;
    }

    printf("%d", x[0]);
    return 0;
}