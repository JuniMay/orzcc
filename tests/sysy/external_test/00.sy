int main(){
    int n = 5;
    int loop = 1;
    int x[5] = {};
    int y[5]  = {1, 2, 3, 4, 5};
    int z[10]  = {10, 9, 8, 7, 6, 5, 4, 3, 2, 1};
    int q = 1, r = 2, t = 3;

    int l = 1;
    while (l <= loop) {
        int k = 0;
        while (k < n) {
            x[k] = q + y[k] * (r * z[k + 1] + t * z[k + 2]);
            k = k+1;
        }
        l = l+1;
    }

    return x[0];
}