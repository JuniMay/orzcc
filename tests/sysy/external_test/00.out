1
11
48